google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddListBasedUserList.java

```diff
@@ -0,0 +1,152 @@
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.userListIds =
+          Arrays.asList(
+              Long.parseLong("INSERT_USER_LIST_ID_HERE"),
+              Long.parseLong("INSERT_USER_LIST_ID_HERE"));
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddListBasedUserList().runExample(googleAdsClient, params.customerId, params.userListIds);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param userListIds the IDs of the lists to be used for the new combination user list.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, List<Long> userListIds) {
+    // Creates the rule operand info targeting the provided list IDs.
+    List<LogicalUserListOperandInfo> logicalUserListOperandInfoList = new ArrayList<>();
+    for (long userListId : userListIds) {
+      String userListResourceName = ResourceNames.userList(customerId, userListId);
+      logicalUserListOperandInfoList.add(
+          LogicalUserListOperandInfo.newBuilder()
+              .setUserList(StringValue.of(userListResourceName))
+              .build());
+    }
+
+    // Creates the UserListLogicalRuleInfo specifying that a user should be added to the new list if
+    // they are present in any of the provided lists.
+    UserListLogicalRuleInfo userListLogicalRuleInfo =
+        UserListLogicalRuleInfo.newBuilder()
+            .setOperator(UserListLogicalRuleOperator.ANY)
```

Comment:
How about adding some optional parenthesis here, to illustrate the various different operators? This can be quite a powerful tool when used well, would be nice to showcase this a little.

Reply:
I thought it might be confusing to list conflicting operators (even if commented out), so I have added a detailed comment explaining the options instead.